### ISSUE TITLE
Snapshot picks up wrong React Native info

### ISSUE THREAD

**Issue by TheAlmightyBob** [infinitered/solidarity#181]
This is very similar to #175 , but specifically affecting `solidarity snapshot` now.

The snapshot is generated with something like the following:
```
    "React Native": [
      {
        "rule": "cli",
        "binary": "react-native",
        "semver": "21"
      },
      {
        "rule": "cli",
        "binary": "react-native",
        "semver": "17",
        "line": 2
      }
    ],
```

There are two different entries for `react-native`, both versions are wrong, and they change every time `solidarity snapshot` is run (suggesting that it is again picking up the number from the "scanning folders" execution time).

When then running `solidarity`, I get:
```
✖ react-native: you have '0.52.2', but the project requires '22'
✖ react-native: you have '2.0.1', but the project requires '18'
```
So in that case it is picking up the correct `react-native` version, but it is _also_ picking up the `react-native-cli` version and reporting that as an additional `react-native` version.

**Comment by GantMan** [comment]
This is hopefully solved by v`2.0.1` which I just released.  Please let me know!

**Comment by TheAlmightyBob** [comment]
@GantMan The snapshot has correct versions now but still includes both the `react-native` and `react-native-cli` versions under the label `react-native`:
```
    "React Native": [
      {
        "rule": "cli",
        "binary": "react-native",
        "semver": "2.0.1"
      },
      {
        "rule": "cli",
        "binary": "react-native",
        "semver": "0.52.2",
        "line": 2
      }
    ],
```
Is that by design?

Also, `solidarity` (without `snapshot`) is no longer working for me with this version. I just get:
```
TypeError: Cannot read property '1' of null

✔︎ Solidarity checks valid
⠧ Verifying Android
```

And it spins forever on `verifying Android`. Previous version does not exhibit this problem.

**Comment by GantMan** [comment]
### Part 1: 
Yes I made the decision to put both checks for line 1 and line 2 of react-native's output.  I figured the more info verified the better.  If you think this is overkill, let me know, but I think that knowing the cli is key because `rnpm` versions are very important, and that's tied to what version of the CLI.  I'd want to know if we had different CLIs


### Part 2:
So bummer... I do not get that spin error.  Strange that it would complain about the type.  Make sure you're on 2.0.1 with `yarn solidarity -v`.  My guess is that you're getting something wonky from your environment that is making the code throw up a lil, but now I'm second guessing that if this is a new bug. 

Can you wipe node_modules and reinstall?

If none of that works, I'll screenshare with you and we can debug if you'd like.

**Comment by patrickkempff** [comment]
I can confirm that i have the same problem with v2.0.1

```
TypeError: Cannot read property '1' of null

✔︎ Solidarity checks valid
⠧ Verifying Android
```

**Comment by TheAlmightyBob** [comment]
@patrickkempff @GantMan I split off the TypeError/Verifying issue into #183 and am closing this specific issue w/ the React Native snapshot, which does appear to be fixed in 2.0.1.
